modules/sildisco/src/IdPDisco.php

```diff
@@ -53,10 +53,8 @@ private function getSPEntityIDAndReducedIdpList(): array
         // Before the SimpleSAMLphp 2 upgrade, we added it to the state ourselves by overriding the SAML2.php file
         parse_str(parse_url($_GET['return'], PHP_URL_QUERY), $returnState);
         $state = Auth\State::loadState($returnState['AuthID'], 'saml:sp:sso');
-        if ($state && array_key_exists('SPMetadata', $state)) {
-            $spmd = $state['SPMetadata'];
-            $this->log('Updated SP metadata from ' . $this->spEntityId . ' to ' . $spmd['entityid']);
-        }
+        assert($state && array_key_exists('SPMetadata', $state));
```

Comment:
I don't think we want to depend on PHP's built in `assert()` function, since it is typically only run in debug or test mode, and is disabled in production mode (per [the docs](https://www.php.net/assert)).

If we want to ensure something is asserted every time, it seems to be better to use other/normal PHP code (either an `if` and `throw new Exception`, or  something like WebMozart's "Assert" library).

Reply:
Oh, I didn't realize that. Thanks.